react/features/calendar-sync/actions.native.js

```diff
@@ -0,0 +1,47 @@
+// @flow
+
+import { getDefaultURL } from '../app';
+import { openDialog } from '../base/dialog';
+import { generateRoomWithoutSeparator } from '../welcome';
+
+import { refreshCalendar } from './actions';
+import { addLinkToCalendarEntry } from './functions.native';
+
+import {
+    UpdateCalendarEventDialog
+} from './components';
+
+export * from './actions.any';
+
+/**
+ * Asks confirmation from the user to add a Jitsi link to the calendar event.
+ *
+ * @param {string} eventId - The event id.
+ * @returns {{
+ *     type: OPEN_DIALOG,
+ *     component: React.Component,
+ *     componentProps: (Object | undefined)
+ * }}
+ */
+export function openUpdateCalendarEventDialog(eventId: string) {
+    return openDialog(UpdateCalendarEventDialog, { eventId });
+}
+
+/**
+ * Updates calendar event by generating new invite URL and editing the event
+ * adding some descriptive text and location.
+ *
+ * @param {string} eventId - The event id.
+ * @returns {Function}
+ */
+export function updateCalendarEvent(eventId: string) {
+    return (dispatch: Dispatch<*>, getState: Function) => {
+        const defaultUrl = getDefaultURL(getState);
+        const roomName = generateRoomWithoutSeparator();
+
+        addLinkToCalendarEntry(getState(), eventId, `${defaultUrl}/${roomName}`)
```

Comment:
The one thing I noticed here when testing manually is if the async action to save the event fails, the event is still updated on the calendar list but wasn't actually saved. Is that a case you want to handle right now?

Reply:
Not sure what you mean. If the event fails to save, then there is no link added to the event. The refresh still happens, but just to start from a clean state afterwards, but a link shouldn't appear in the event in these cases. Did you happen to see a link added to an event in the app but not saved in the calendar event in the system calendar?

Reply:
Yeah, that's what I was seeing. I clicked the + button and confirmed and a linked showed up on the meeting. I went to my google calendar and noticed no link. I refreshed the calendar list and the link disappeared.